HELP.md

```diff
@@ -0,0 +1,761 @@
+  within a contract directly (not through a promise).
+- `pub(crate) fn` also will make a method internal. It's helpful to use it when you have a method in a different module.
+
+```rust
+#[near_bindgen]
+impl Contract {
+    pub fn increment(&mut self) {
+        self.internal_increment();
+    }
+
+    fn internal_increment(&mut self) {
+        self.counter += 1;
+    }
+}
+```
+
+Another way of not exporting methods is by having a separate `impl Contract` section, that is not marked with `#[near_bindgen]`.
+
+```rust
+#[near_bindgen]
+impl Contract {
+    pub fn increment(&mut self) {
+        self.internal_increment();
+    }
+}
+
+impl Contract {
+    /// This methods is still not exported.
+    pub fn internal_increment(&mut self) {
+        self.counter += 1;
+    }
+}
+```
+
+## Callbacks
+
+Callbacks have to be public methods exported from the contract, they needs to be called using a function call.
+
+If you're using callbacks, makes sure you check the predecessor to avoid someone else from calling it.
+
+There is an macro decorator `#[private]` that checks that the current account ID is equal to the predecessor account ID.
+
+```rust
+#[near_bindgen]
+impl Contract {
+    #[private]
+    pub fn resolve_transfer(&mut self) {
+        env::log(b"This is a callback");
+    }
+}
+```
+
+This is equivalent to:
+
+```rust
+#[near_bindgen]
+impl Contract {
+    pub fn resolve_transfer(&mut self) {
+        if env::current_account_id() != env::predecessor_account_id() {
+            near_sdk::env::panic(b"Method resolve_transfer is private");
+        }
+        env::log(b"This is a callback");
+    }
+}
+```
+
+## Integer JSON types
+
+NEAR Protocol currently expects contracts to support JSON serialization. JSON can't handle large integers (above 2**53 bits).
+That's why you should use helper classes from the `json_types` in `near_sdk` for `u64` and `u128`.
+We provide types `U64` and `U128`, that wraps the integer into a struct and implements JSON serialization and
+deserialization as a base-10 strings.
+
+You can convert from `U64` to `u64` and back using `std::convert::Into`, e.g.
+
+```rust
+#[near_bindgen]
+impl Contract {
+    pub fn mult(&self, a: U64, b: U64) -> U128 {
+        let a: u64 = a.into();
+        let b: u64 = b.into();
+        let product = u128::from(a) * u128::from(b);
+        product.into()
+    }
+}
+```
+
+You can also access inner values and using `.0` and `U128(5)`, e.g.
+
+```rust
+#[near_bindgen]
+impl Contract {
+    pub fn sum(&self, a: U128, b: U128) -> U128 {
+        U128(a.0 + b.0)
+    }
+}
+```
+
+## `Base64VecU8` JSON type
+
+Often the contract needs to receive or return a binary data.
+Encoding a `Vec<u8>` with JSON will lead to an integer array, e.g. `[110, 101, 97, 114]`
+This is inefficient in both compute and space.
+
+`Base64VecU8` is a wrapper on top of `Vec<u8>` that allows to pass it as arguments or result.
```

Comment:
Why is this more efficient in either compute or space? 

We store it internally as `Vec<u8>` anyhow, so the space consideration seems irrelevant.

And for compute on in/out, base64 is [about 33% larger than binary](https://stackoverflow.com/questions/33811673/why-the-size-of-base64-encoded-string-is-larger-than-the-original-file). How then is serializing & deserializing base64 more compute-efficient than using binary?

If neither of these are true, is there some other situation that should make people reach for `Base64VecU8`?